### PR TITLE
Replace methods with properties where appropriate

### PR DESCRIPTION
**Pull request by alexjpwalker**
## What is the goal of this PR?

Because client-nodejs is based on Client Java, we were using methods in a lot of places where a property would be more fitting. So, for example, we've changed `Concept.getType()` to just `Concept.type`, and `Type.getLabel()` to just `Type.label`.

## What are the changes implemented in this PR?

Replace methods with properties where appropriate

### CODE REVIEW

File: api/answer/ConceptMap.ts [hunk 1]
```diff
@@ -19,24 +19,21 @@
  * under the License.
  */
 
-
 import { Concept } from "../concept/Concept";
 
 export interface ConceptMap {
 
-    map(): Map<string, Concept>;
+    readonly map: Map<string, Concept>;
```
Comment (alexjpwalker) [inline api/answer/ConceptMap.ts:26]: This syntax specifies that this property is read-only, i.e. you can't assign anything to it.

File: api/connection/TypeDBCredential.ts [hunk 2]
```diff
@@ -40,16 +40,15 @@ export class TypeDBCredential {
         this._tlsRootCAPath = tlsRootCAPath;
     }
 
-    public username(): string {
```
Comment (alexjpwalker) [inline api/connection/TypeDBCredential.ts:43]: In TypeScript, `public` is the default modifier

File: concept/thing/EntityImpl.ts [hunk 3]
```diff
@@ -28,10 +28,10 @@ import { EntityTypeImpl, ThingImpl } from "../../dependencies_internal";
 
 export class EntityImpl extends ThingImpl implements Entity {
 
-    private _type: EntityType;
+    private readonly _type: EntityType;
 
-    constructor(iid: string, isInferred: boolean, type: EntityType) {
-        super(iid, isInferred);
+    constructor(iid: string, inferred: boolean, type: EntityType) {
+        super(iid, inferred);
```
Comment (alexjpwalker) [inline concept/thing/EntityImpl.ts:34]: In JavaScript, it's considered an anti-pattern for boolean flags to start with the word "is". (This is in stark contrast to Java, where all booleans are expected to start with a verb)

File: logic/RuleImpl.ts [hunk 4]
```diff
@@ -61,44 +61,41 @@ export class RuleImpl implements Rule {
...
 }
 
-export class RemoteRuleImpl extends RuleImpl implements RemoteRule {
-    private _transaction: TypeDBTransaction.Extended;
-
-    constructor(transaction: TypeDBTransaction.Extended, label: string, when: string, then: string) {
-        super(label, when, then);
-        this._transaction = transaction;
-    }
+export namespace RuleImpl {
```
Comment (alexjpwalker) [inline logic/RuleImpl.ts:66]: whoops, I forgot to un-flatten this one in the last PR

File: api/connection/TypeDBOptions.ts [hunk 5]
```diff
@@ -48,43 +51,130 @@ namespace Opts {
...
+export class TypeDBOptions implements Opts.Core {
+
+    private _infer: boolean;
+    private _traceInference: boolean;
+    private _explain: boolean;
+    private _parallel: boolean;
+    private _prefetchSize: number;
+    private _prefetch: boolean;
+    private _sessionIdleTimeoutMillis: number;
+    private _schemaLockAcquireTimeoutMillis: number;
```
Comment (flyingsilverfin) [inline api/connection/TypeDBOptions.ts:72]: interesting how come you decided to declare everything?

File: api/connection/TypeDBOptions.ts [hunk 6]
```diff
@@ -48,43 +51,130 @@ namespace Opts {
...
+    }
+
+    set explain(value: boolean) {
+        this._explain = value;
+    }
+
+    get parallel() {
+        return this._parallel;
+    }
+
```
Comment (flyingsilverfin) [inline api/connection/TypeDBOptions.ts:113]: why dont' we use the properties approach here too then?

Comment (alexjpwalker) [reply api/connection/TypeDBOptions.ts:113]: This _is_ the properties approach. You declare a private backing field, and a `get` method as the property getter, and in this case specifically, we also declare a `set` method as the property setter.

**Review by flyingsilverfin — APPROVED** [review]

**Pull request merged**